drf_renderer_xlsx/fields.py

```diff
@@ -0,0 +1,185 @@
+	sanitize = True
+
+	def __init__(self, key, value, field: Field, style: NamedStyle, mapping: Union[str, Callable]):
+		self.key = key
+		self.original_value = value
+		self.drf_field = field
+		self.style = style or NamedStyle()
+		self.mapping = mapping
+		self.value = self.init_value(value)
+
+	def init_value(self, value):
+		return value
+
+	def sanitize_value(self, value):
+		# prepend ' if value is starting with possible malicious char
+		if self.sanitize and value:
+			str_value = str(value)
+			str_value = ILLEGAL_CHARACTERS_RE.sub('', str_value)  # remove ILLEGAL_CHARACTERS so it doesn't crash
+			return "'" + str_value if str_value.startswith(ESCAPE_CHARS) else str_value
+		return value
+
+	def custom_mapping(self):
+		if type(self.mapping) is str:
+			return self.value.get(self.mapping)
+		elif callable(self.mapping):
+			return self.mapping(self.value)
+		return self.value
+
+	def prep_value(self) -> Any:
+		return self.value
+
+	def prep_cell(self, cell: Cell):
+		# We cannot apply the whole style directly, otherwise we cannot override any part of it
+		cell.font = self.style.font
+		cell.fill = self.style.fill
+		cell.alignment = self.style.alignment
+		cell.border = self.style.border
+		cell.number_format = self.style.number_format
+
+	def cell(self, ws: Worksheet, row, column) -> Cell:
+		# If we have a custom mapping use it and done. If not prep value for output
+		value = self.sanitize_value(self.custom_mapping() if self.mapping else self.prep_value())
+		cell: Cell = ws.cell(row, column, value)
+		self.prep_cell(cell)
+		return cell
+
+
+class XLSXNumberField(XLSXField):
+	sanitize = False
+
+	def __init__(self, number_format, **kwargs):
+		self.number_format = number_format
+		super().__init__(**kwargs)
+
+	def init_value(self, value):
+		try:
+			if isinstance(self.drf_field, IntegerField) and type(value) != int:
+				return int(value)
+			elif isinstance(self.drf_field, FloatField) and type(value) != float:
+				return float(value)
+			elif isinstance(self.drf_field, DecimalField) and type(value) != Decimal:
+				return Decimal(value)
+		except:
+			pass
+		return value
+
+	def prep_cell(self, cell: Cell):
+		super().prep_cell(cell)
+		if self.number_format:
+			cell.number_format = self.number_format
+		elif isinstance(self.drf_field, IntegerField):
+			cell.number_format = get_setting('INTEGER_FORMAT') or FORMAT_NUMBER
+		else:
+			cell.number_format = get_setting('DECIMAL_FORMAT') or FORMAT_NUMBER_00
+
+
+class XLSXDateField(XLSXField):
+	sanitize = False
+
+	def __init__(self, date_format, **kwargs):
+		self.date_format = date_format
+		super().__init__(**kwargs)
+
+	def _parse_date(self, value, setting_format, iso_parse_func):
+		# Parse format is DRF output format: DATETIME_FORMAT, DATE_FORMAT or TIME_FORMAT
+		parse_format = getattr(drf_settings, setting_format)
+		# Use the provided iso parse function for the special case ISO_8601
+		if parse_format.lower() == ISO_8601:
+			return iso_parse_func(value)
```

Comment:
This leads to `TypeError: can't subtract offset-naive and offset-aware datetimes` on my dataset

Crash happens in `openpyxl.utils.datetime.to_excel`. `dt == datetime.datetime(2022, 2, 16, 9, 34, 4, 928350, tzinfo=datetime.timezone(datetime.timedelta(seconds=3600), '+0100'))`

Reply:
I think I fixed it, although I was getting a slightly different error message. 
could you check now?

Reply:
removing timezone works good enough :D